diff --git a/src/index.d.ts b/src/index.d.ts
index 58c0b89..69f60aa 100644
--- a/src/index.d.ts
+++ b/src/index.d.ts
@@ -1,5 +1,45 @@
+export interface BaseOperation {
+  path: string;
+}
+
+export interface AddOperation<T = any> extends BaseOperation{
+  op: 'add';
+  value: T;
+};
+
+export interface RemoveOperation extends BaseOperation{
+  op: 'remove';
+};
+
+export interface ReplaceOperation<T extends any> extends BaseOperation {
+  op: 'replace';
+  value: T;
+};
+
+export interface MoveOperation extends BaseOperation {
+  op: 'move';
+  from: string;
+};
+
+export interface CopyOperation extends BaseOperation {
+  op: 'copy';
+  from: string;
+}
+export interface TestOperation<T> extends BaseOperation {
+  op: 'test';
+  value: T;
+}
+
+export type Operation = AddOperation | RemoveOperation | ReplaceOperation | MoveOperation | CopyOperation | TestOperation
+
+export type Patch = Operation[]
+
+export interface JSONPatchFormatter {
+  format(delta: Delta | undefined, original?: any): Patch;
+}
+
 export interface Formatter {
-    format(delta: Delta, original: any): string;
+  format(delta: Delta, original: any): string;
 }
 
 export interface HtmlFormatter extends Formatter {
@@ -190,6 +230,7 @@ export const formatters: {
   annotated: Formatter;
   console: Formatter;
   html: HtmlFormatter;
+  jsonpatch: JSONPatchFormatter;
 };
 
 export const console: Formatter
diff --git a/dist/jsondiffpatch.cjs.js b/dist/jsondiffpatch.cjs.js
index 1e53d97..86e0dca 100644
--- a/dist/jsondiffpatch.cjs.js
+++ b/dist/jsondiffpatch.cjs.js
@@ -5,7 +5,7 @@ Object.defineProperty(exports, '__esModule', { value: true });
 function _interopDefault (ex) { return (ex && (typeof ex === 'object') && 'default' in ex) ? ex['default'] : ex; }
 
 var dmp = _interopDefault(require('diff-match-patch'));
-var chalk = _interopDefault(require('chalk'));
+var chalk = undefined;
 
 var _typeof = typeof Symbol === "function" && typeof Symbol.iterator === "symbol" ? function (obj) {
   return typeof obj;
diff --git a/dist/jsondiffpatch.esm.js b/dist/jsondiffpatch.esm.js
index e60696c..b96dc5e 100644
--- a/dist/jsondiffpatch.esm.js
+++ b/dist/jsondiffpatch.esm.js
@@ -1,5 +1,7 @@
 import dmp from 'diff-match-patch';
-import chalk from 'chalk';
+// import chalk from 'chalk';
+
+const chalk = undefined
 
 var _typeof = typeof Symbol === "function" && typeof Symbol.iterator === "symbol" ? function (obj) {
   return typeof obj;
